service/interpreter/continueAsNewer.go

```diff
@@ -110,27 +113,56 @@ func LoadInternalsFromPreviousRun(
 	return &resp, nil
 }
 
+func (c *ContinueAsNewer) GetSnapshot() service.ContinueAsNewDumpResponse {
+	localStateExecutionToResumeMap := map[string]service.StateExecutionResumeInfo{}
+	for key, state := range c.StateExecutionToResumeMap {
+		localStateExecutionToResumeMap[key] = state
+	}
+	for _, value := range c.stateRequestQueue.GetAllStateResumeRequests() {
+		localStateExecutionToResumeMap[value.StateExecutionId] = value
+	}
+	return service.ContinueAsNewDumpResponse{
+		InterStateChannelReceived:  c.interStateChannel.GetAllReceived(),
+		SignalsReceived:            c.signalReceiver.GetAllReceived(),
+		StateExecutionCounterInfo:  c.stateExecutionCounter.Dump(),
+		DataObjects:                c.persistenceManager.GetAllDataObjects(),
+		SearchAttributes:           c.persistenceManager.GetAllSearchAttributes(),
+		StatesToStartFromBeginning: c.stateRequestQueue.GetAllStateStartRequests(),
+		StateExecutionsToResume:    localStateExecutionToResumeMap,
+		StateOutputs:               c.outputCollector.GetAll(),
+		StaleSkipTimerSignals:      c.timerProcessor.Dump(),
+	}
+}
+
 func (c *ContinueAsNewer) SetQueryHandlersForContinueAsNew(ctx UnifiedContext) error {
-	return c.provider.SetQueryHandler(ctx, service.ContinueAsNewDumpQueryType, func() (*service.ContinueAsNewDumpResponse, error) {
-		localStateExecutionToResumeMap := map[string]service.StateExecutionResumeInfo{}
-		for key, state := range c.StateExecutionToResumeMap {
-			localStateExecutionToResumeMap[key] = state
-		}
-		for _, value := range c.stateRequestQueue.GetAllStateResumeRequests() {
-			localStateExecutionToResumeMap[value.StateExecutionId] = value
-		}
-		return &service.ContinueAsNewDumpResponse{
-			InterStateChannelReceived:  c.interStateChannel.GetAllReceived(),
-			SignalsReceived:            c.signalReceiver.GetAllReceived(),
-			StateExecutionCounterInfo:  c.stateExecutionCounter.Dump(),
-			DataObjects:                c.persistenceManager.GetAllDataObjects(),
-			SearchAttributes:           c.persistenceManager.GetAllSearchAttributes(),
-			StatesToStartFromBeginning: c.stateRequestQueue.GetAllStateStartRequests(),
-			StateExecutionsToResume:    localStateExecutionToResumeMap,
-			StateOutputs:               c.outputCollector.GetAll(),
-			StaleSkipTimerSignals:      c.timerProcessor.Dump(),
-		}, nil
-	})
+	return c.provider.SetQueryHandler(ctx, service.ContinueAsNewDumpQueryType,
+		func(request iwfidl.WorkflowDumpRequest) (*iwfidl.WorkflowDumpResponse, error) {
+			wholeSnapshot := c.GetSnapshot()
+			wholeData, err := json.Marshal(wholeSnapshot)
+			if err != nil {
+				return nil, err
+			}
+			checksum := md5.Sum(wholeData)
+			pageSize := int32(service.DefaultContinueAsNewPageSizeInBytes)
+			if request.PageSizeInBytes > 0 {
+				pageSize = request.PageSizeInBytes
+			}
+			lenInDouble := float64(len(wholeData))
```

Comment:
Nit: no need to declare this since it's only used in line 151. I also believe they could be float32s since you are really dividing ints and just doing type conversion

Reply:
I think having two lines is better for reading. 
The compiler should be smart enough to optimize this to save a local varable. 

float64 is required by the library. Golang doesn't allow int32 to divided by float like Java. 

![Screenshot 2024-12-09 at 3 42 27 PM](https://github.com/user-attachments/assets/a69ef5e9-3201-4e79-97f4-69d4309ab984)

Reply:
<img width="292" alt="Screenshot 2024-12-10 at 9 53 45 AM" src="https://github.com/user-attachments/assets/5b94c1af-e56c-4ea7-9533-4abb96e2e7b7">

Okay, I see that Ceil requires float64, interesting... TIL